redis/src/cluster_async/mod.rs

```diff
@@ -1688,6 +1687,192 @@ impl Connect for MultiplexedConnection {
+    warn!(
+        "New IP was found for node {:?}: 
+                new connection IP = {:?}, previous connection IP = {:?}",
+        addr, new_ip, prev_ip
+    );
+}
+
+fn warn_management_conn_faild(addr: &str, err: RedisError) {
+    warn!(
+        "Failed to create management connection for node {:?}. Error: {:?}",
+        addr, err
+    );
+}
+
+fn create_or_change_async_node<C>(
+    user_conn: C,
+    management_conn: Option<C>,
+    ip: Option<IpAddr>,
+    node: Option<AsyncClusterNode<C>>,
+) -> AsyncClusterNode<C>
+where
+    C: ConnectionLike + Connect + Send + Sync + 'static + Clone,
+{
+    let user_conn = async { user_conn }.boxed().shared();
+    let management_conn =
+        management_conn.map(|management_conn| async { management_conn }.boxed().shared());
+    let mut new_node;
+    if let Some(node) = node {
+        new_node = node;
+        new_node.user_connection = user_conn;
+        new_node.management_connection = management_conn;
+        new_node.ip = ip;
+    } else {
+        new_node = AsyncClusterNode::new(user_conn, management_conn, ip);
+    }
+    new_node
+}
+
+async fn connect_and_check_all_connections<C>(
+    addr: &str,
+    params: ClusterParams,
+    socket_addr: Option<SocketAddr>,
+    node: Option<AsyncClusterNode<C>>,
+) -> RedisResult<AsyncClusterNode<C>>
+where
+    C: ConnectionLike + Connect + Send + Sync + 'static + Clone,
+{
+    match future::join(
+        create_connection(addr, params.clone(), socket_addr),
+        create_connection(addr, params.clone(), socket_addr),
+    )
+    .await
+    {
+        (Ok(conn_1), Ok(conn_2)) => {
+            // Both connections were successfully established
+            let (mut user_conn, user_ip): (C, Option<IpAddr>) = conn_1;
+            let (mut management_conn, management_ip): (C, Option<IpAddr>) = conn_2;
+            if user_ip != management_ip && user_ip.is_some() {
+                // use only the connection with the latest IP address
+                warn_mismatch_ip(addr, user_ip, management_ip);
+                if is_dns_changed(addr, &user_ip.unwrap()).await {
+                    // The user_ip is incorrect. Use the cerated `management_conn` for the user connection
+                    user_conn = management_conn;
+                    setup_user_connection(&mut user_conn, params).await?;
+                    Ok(create_or_change_async_node(
+                        user_conn,
+                        None,
+                        management_ip,
+                        node,
+                    ))
+                } else {
+                    // The user_ip is correct. Use the user connetion and drop the management connection
+                    setup_user_connection(&mut user_conn, params).await?;
+                    Ok(create_or_change_async_node(user_conn, None, user_ip, node))
+                }
+            } else {
+                // Both connections share the same IP. Setup both
```

Comment:
this can also happen if `user_ip != management_ip && user_ip.id_none()`

Reply:
the user/management IP will be none only if it's a unix domain socket. In a UDS case we can't have a dns change or something like it that would cause two connections to different addresses. So if user_ip is none, management_ip is none. 
I will change the comment to:
```
Both connections share the same address. Setup both
```
And add a comment that explains the UDS and none thing

Reply:
```
if user_ip.is_some() && user_ip != management_ip {
...           
 } else {
                // If the returned IP is None, it implies connection via a Unix domain socket. With UDS, connecting to different addresses isn't possible.
                // Therefore, in this scenario, both connections share the same address. We shall set up both connections accordingly.

Reply:
nvm. changed to 
```
if user_ip == management_ip { 
..
} else {
... 
}